### PR TITLE
Fix subscription observers firing on launch

### PR DESCRIPTION
**Pull request by emawby**
The OSSubscriptionObserver was firing 3 times on launch for a fresh install.

1. For prompting permission
1. getting the APNs token
1. For registering the user

Now it will only fire after register user has been called and either succeeded or failed. For subsequent launches we rely on the `isOnSessionSuccessfulForCurrentState` boolean.

## 
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-ios-sdk/853)

### CODE REVIEW

**Comment by emawby** [comment]
The branch does not include the set external user id unit test fix

File: iOS_SDK/OneSignalSDK/Source/OneSignal.m [hunk 1]
```diff
@@ -1536,6 +1535,17 @@ + (void)setIsOnSessionSuccessfulForCurrentState:(BOOL)value {
     isOnSessionSuccessfulForCurrentState = value;
 }
 
+static BOOL _registerUserFinished = false;
++ (BOOL)isRegisterUserFinished {
+    if (_registerUserFinished || isOnSessionSuccessfulForCurrentState) {
+        NSLog(@"ECM register finished");
```
Comment (jkasten2) [inline iOS_SDK/OneSignalSDK/Source/OneSignal.m:1541]: Should use `onesignalLog` instead of `NSLog`

File: iOS_SDK/OneSignalSDK/Source/OneSignal.m [hunk 2]
```diff
@@ -1536,6 +1535,17 @@ + (void)setIsOnSessionSuccessfulForCurrentState:(BOOL)value {
...
+    } else {
+        NSLog(@"ECM register NOT finished");
```
Comment (jkasten2) [inline iOS_SDK/OneSignalSDK/Source/OneSignal.m:1543]: Should use `onesignalLog` instead of `NSLog`

**Review by jkasten2 — CHANGES_REQUESTED** [review]
* `NSLog` should not be used
* Missing test to cover new logic

File: iOS_SDK/OneSignalSDK/Source/OneSignal.m (hunk 1 above)
Comment (emawby) [reply iOS_SDK/OneSignalSDK/Source/OneSignal.m]: my fixup didn't get force pushed I removed the logs

File: iOS_SDK/OneSignalSDK/Source/OneSignal.m (hunk 2 above)
Comment (emawby) [reply iOS_SDK/OneSignalSDK/Source/OneSignal.m]: my fixup didn't get force pushed I removed the logs

**Review by jkasten2 — APPROVED** [review]

**Pull request merged**